test/unittests/util/test_format_en.py

```diff
@@ -0,0 +1,370 @@
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_ampm=True),
+                         "1:00 PM")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_24hour=True),
+                         "13:00")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_24hour=True, use_ampm=True),
+                         "13:00")
+        self.assertEqual(nice_time(dt, use_24hour=True,
+                                   use_ampm=True),
+                         "thirteen hundred")
+        self.assertEqual(nice_time(dt, use_24hour=True,
+                                   use_ampm=False),
+                         "thirteen hundred")
+
+        dt = datetime.datetime(2017, 1, 31,
+                               13, 2, 3)
+        self.assertEqual(nice_time(dt),
+                         "one oh two")
+        self.assertEqual(nice_time(dt, use_ampm=True),
+                         "one oh two p.m.")
+        self.assertEqual(nice_time(dt, speech=False),
+                         "1:02")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_ampm=True),
+                         "1:02 PM")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_24hour=True),
+                         "13:02")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_24hour=True, use_ampm=True),
+                         "13:02")
+        self.assertEqual(nice_time(dt, use_24hour=True,
+                                   use_ampm=True),
+                         "thirteen zero two")
+        self.assertEqual(nice_time(dt, use_24hour=True,
+                                   use_ampm=False),
+                         "thirteen zero two")
+
+        dt = datetime.datetime(2017, 1, 31,
+                               0, 2, 3)
+        self.assertEqual(nice_time(dt),
+                         "twelve oh two")
+        self.assertEqual(nice_time(dt, use_ampm=True),
+                         "twelve oh two a.m.")
+        self.assertEqual(nice_time(dt, speech=False),
+                         "12:02")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_ampm=True),
+                         "12:02 AM")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_24hour=True),
+                         "00:02")
+        self.assertEqual(nice_time(dt, speech=False,
+                                   use_24hour=True, use_ampm=True),
+                         "00:02")
+        self.assertEqual(nice_time(dt, use_24hour=True,
+                                   use_ampm=True),
+                         "zero zero zero two")
+        self.assertEqual(nice_time(dt, use_24hour=True,
+                                   use_ampm=False),
+                         "zero zero zero two")
```

Comment:
is the time really spoken this way? with the zero sounds awkward to me, but I am not a antive english speaker

Reply:
Yup apparently it is google tells me xD